clang/include/clang/Driver/Options.td

```diff
@@ -1822,6 +1822,13 @@ def fsycl_add_targets_EQ : CommaJoined<["-"], "fsycl-add-targets=">, Flags<[Driv
   HelpText<"Specify comma-separated list of triple and device binary image pairs to add to the final SYCL binary">;
 def fsycl_link_targets_EQ : CommaJoined<["-"], "fsycl-link-targets=">, Flags<[DriverOption, CC1Option, CoreOption]>,
   HelpText<"Specify comma-separated list of triples SYCL offloading targets to produce linked device images">;
+def fsycl_device_code_split_EQ : Joined<["-"], "fsycl-device-code-split=">,
+   Flags<[CC1Option, CoreOption]>, HelpText<"Perform SYCL device code split: per_kernel (device code module is "
+  "created for each SYCL kernel) | per_source (device code module is created for each source (translation unit)). "
+  "Default is no device code split - all kernels go into a single module`">, Values<"per_source,per_kernel">;
```

Comment:
Should default value be one of the possible options? E.g. -fsycl-device-code-split=off.

Reply:
I suggest add this in a separate patch.

Reply:
I suggest add it in next commits.

Reply:
@bader, please approve the PR if you're OK with this.